### PR TITLE
Add Dynamic Notch Debug Mode and others

### PR DESCRIPTION
**Pull request by McGiverGim**
This adds support for the `debug_mode` dynamic notch, more info here:

https://github.com/betaflight/betaflight/pull/3074

This code lets show the info, and works ok, but maybe need some adjustments in scale, power or units. Any help to refine that will be welcome.

### CODE REVIEW

**Comment by McGiverGim** [comment]
@rav-rav with this PR your debug values are shown in the log viewer, but I don't know anything about units or what represents each value, so maybe some of them will need changes.
There're three parts:
- The first, the name that appears. It must be as short as possible, for example:
```
'DEBUG_FFT' : {
    		                'debug[all]':'Debug FFT',
    		                'debug[0]':'gyro_preFFT[roll]',
    		                'debug[1]':'gyro_postFFT[roll]',
    		                'debug[2]':'Res. band pass',
    		                'debug[3]':'Weighted center', 
```
- The second the unit conversion. I've used the same as for the static NOTCH filter for the debug 0, 1 and 2 of `DEBUG_FFT` (conversion to deg/s) and let the fixed value for the others.
- Last, to "adjust" to the graph, there's some parameters, for example:
```
                   case 'DEBUG_FFT':
                        return {
                            offset: 0,
                            power: 0.25,
                            inputRange: (2.0e-3 * Math.PI/180) / sysConfig.gyroScale,
                            outputRange: 1.0
                        };      
```
again used the same as NOTCH except for the `DEBUG_FFT_TIME` that I fixed the input range (it was so small).

If you have time, can you review the code to see if I need to make some changes to some of the values? As I said, I don't have any knowledge about filters, gyros, etc.

Thanks!

File: js/flightlog_fields_presenter.js [hunk 1]
```diff
@@ -215,7 +215,28 @@ function FlightLogFieldPresenter() {
...
-    		}
+    		},
+    		'DEBUG_FFT' : {
+    		                'debug[all]':'Debug FFT',
+    		                'debug[0]':'gyro_preFFT[roll]',
+    		                'debug[1]':'gyro_postFFT[roll]',
+    		                'debug[2]':'Res. band pass',
+    		                'debug[3]':'Weighted center',    		
+    		},
+            'DEBUG_FFT_TIME' : {
```
Comment (rav-rav) [inline js/flightlog_fields_presenter.js]: This needs to be switched with DEBUG_FFT_FREQ.

File: js/flightlog_fields_presenter.js [hunk 2]
```diff
@@ -215,7 +215,28 @@ function FlightLogFieldPresenter() {
 							'debug[1]':'debug[1]',
 							'debug[2]':'debug[2]',
 							'debug[3]':'debug[3]',    		
-    		}
+    		},
+    		'DEBUG_FFT' : {
+    		                'debug[all]':'Debug FFT',
+    		                'debug[0]':'gyro_preFFT[roll]',
```
Comment (rav-rav) [inline js/flightlog_fields_presenter.js]: FFT (fast fourier transform) is the process used to calculate the frequency spectrum. But what is actually debugged here is the dynamic notch filter. (Probably that should be renamed...)
I'd prefer these names:
0: gyro_raw[roll]
1: gyro_dyn_notch[roll]
2: gyro_bpf[roll]
3: fft center index[roll]

File: js/graph_config.js [hunk 3]
```diff
@@ -415,6 +415,27 @@ GraphConfig.load = function(config) {
...
+                            offset: 0,
+                            power: 0.25,
+                            inputRange: (2.0e-3 * Math.PI/180) / sysConfig.gyroScale,
+                            outputRange: 1.0
+                        };     
+                    case 'DEBUG_FFT_TIME':
+                        return {
+                            offset: 0,
+                            power: 0.25,
+                            inputRange: 30,
```
Comment (rav-rav) [inline js/graph_config.js]: This might change and go up above 30. 100 would be a better value.
It should also be displayed as linear possible. (No power?)

**Comment by rav-rav** [comment]
Looking good!
I added some comments to the code.
Units are deg/s for DEBUG_FFT, Hz for DEBUG_FFT_FREQ and µs for DEBUG_FFT_TIME

File: js/flightlog_fields_presenter.js (hunk 2 above)
Comment (McGiverGim) [reply js/flightlog_fields_presenter.js]: Ok, perfect. I will change it.

File: js/graph_config.js (hunk 3 above)
Comment (McGiverGim) [reply js/graph_config.js]: I don't know. But I will try ;)

File: js/flightlog_fields_presenter.js (hunk 1 above)
Comment (McGiverGim) [reply js/flightlog_fields_presenter.js]: I don't understand what are you wanting to say with "switched" :(

**Comment by McGiverGim** [comment]
Changes made as requested.

File: js/flightlog_fields_presenter.js [hunk 4]
```diff
@@ -441,9 +462,15 @@ function FlightLogFieldPresenter() {
...
                     }
                 case 'SCHEDULER':
-                        return value.toFixed(0) + "\u03BCS";
+                    return value.toFixed(0) + "\u03BCS";
                 case 'STACK':
                     return value.toFixed(0);
+                case 'DEBUG_FFT':
+                    return Math.round(flightLog.gyroRawToDegreesPerSecond(value)) + "deg/s";               
+                case 'DEBUG_FFT_TIME':
+                    return value.toFixed(0) + "Hz";
```
Comment (ghost) [inline js/flightlog_fields_presenter.js]: as @rav-rav mentioned must be µs not Hz

File: js/flightlog_fields_presenter.js [hunk 5]
```diff
@@ -441,9 +462,15 @@ function FlightLogFieldPresenter() {
...
+                case 'DEBUG_FFT_FREQ':
+                    return value.toFixed(0) + "\u03BCS";
```
Comment (ghost) [inline js/flightlog_fields_presenter.js]: as @rav-rav also mentioned must be Hz

Comment (McGiverGim) [reply js/flightlog_fields_presenter.js]: I inverted it by error. Thanks for your review!

File: js/flightlog_fields_presenter.js [hunk 6]
```diff
@@ -215,7 +215,28 @@ function FlightLogFieldPresenter() {
...
+                            'debug[3]':'Not used',           
+            },
+            'DEBUG_FFT_FREQ' : {
+                            'debug[all]':'Debug FFT FREQ',
+                            'debug[0]':'Active calc step',
+                            'debug[1]':'Step duration',
+                            'debug[2]':'Aditional steps',
+                            'debug[3]':'Not used',           
+            }
     		};
```
Comment (ghost) [inline js/flightlog_fields_presenter.js]: as @rav-rav mentioned must be switched like this:
 
```
            'DEBUG_FFT_TIME' : {
                             'debug[all]':'Debug FFT TIME',
                             'debug[0]':'Active calc step',
                             'debug[1]':'Step duration',
                             'debug[2]':'Additional steps',
                             'debug[3]':'Not used',           
             },
             'DEBUG_FFT_FREQ' : {
                             'debug[all]':'Debug FFT FREQ',
                             'debug[0]':'center_freq[roll]',
                             'debug[1]':'center_freq[pitch]',
                             'debug[2]':'center_freq[yaw]',
                             'debug[3]':'Not used',           
             }
```

Comment (McGiverGim) [reply js/flightlog_fields_presenter.js]: Ok, understood.

**Comment by McGiverGim** [comment]
Changes done. Thanks!

File: js/flightlog_fields_presenter.js [hunk 7]
```diff
@@ -215,7 +215,28 @@ function FlightLogFieldPresenter() {
...
+    		},
+    		'DEBUG_FFT' : {
+    		                'debug[all]':'Debug FFT',
+    		                'debug[0]':'gyro_raw[roll]',
+    		                'debug[1]':'gyro_dyn_notch[roll]',
+    		                'debug[2]':'gyro_bpf[roll]',
+    		                'debug[3]':'fft_center_index[roll]',    		
+    		},
+            'DEBUG_FFT_TIME' : {
+                            'debug[all]':'Debug FFT FREQ',
```
Comment (ghost) [inline js/flightlog_fields_presenter.js]: should be `'debug[all]':'Debug FFT TIME',`

File: js/flightlog_fields_presenter.js [hunk 8]
```diff
@@ -215,7 +215,28 @@ function FlightLogFieldPresenter() {
...
+                            'debug[0]':'Active calc step',
+                            'debug[1]':'Step duration',
+                            'debug[2]':'Aditional steps',
+                            'debug[3]':'Not used',           
+            },
+            'DEBUG_FFT_FREQ' : {
+                            'debug[all]':'Debug FFT TIME',
```
Comment (ghost) [inline js/flightlog_fields_presenter.js]: ... and this should be `'debug[all]':'Debug FFT FREQ',`

Comment (McGiverGim) [reply js/flightlog_fields_presenter.js]: Buahhh! Next time I will recheck things  3 times... Promised! 😑
Thanks again for your time!

**Comment by McGiverGim** [comment]
Changes done. Thanks again!

**Pull request merged**